### PR TITLE
Dnenov packageintegration mypackages

### PR DESCRIPTION
**Pull request by dnenov**
Not to be merged. Currently broken down in separate PRs.

### Purpose

This is a follow-up PR on the [package integration PR](https://github.com/DynamoDS/Dynamo/pull/14075). It builds on the continuous design guides as per [this Figma link](https://www.figma.com/file/MJh76m7bbafE3sjwLUTZhZ/Dynamo-package-manager-update?type=design&node-id=157-166616&t=L8WJUAlMlyYSPl5n-0).

Even though a lot of areas have been improved, this is still work in progress. With the main skeleton of the solution in place, the major directions of work are:
- design changes in Search for packages tab
- design changes in Publish a package tab
- design changes in My packages tab
- design changes in Publish version tab

#### Screenshots

**01 loading screen**
01_loading screen

**02 discard loading timeout**
02_discard loading timeout

**03 package details extension interaction** 
03_package details extension

**04 package manager contents**
04_package manager contents

**05 package manager to preferences interaction**
05_package manager to preferences

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Release Notes

Progression work on the Package Manager
- interactions between different parts of the package manager and between package manager and preferences screen 
- added empty, loading, idle screens
- work on the PacakgeDetailsExtension to allow hosting inside the PackageManager
- misc refactor

### Reviewers

@reddyashish
@QilongTang

### FYIs

@Amoursol

### CODE REVIEW

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 // associated with an assembly.
 [assembly: AssemblyCompany("Autodesk, Inc")]
 [assembly: AssemblyProduct("Dynamo")]
-[assembly: AssemblyCopyright("Copyright © Autodesk, Inc 2023")]
+[assembly: AssemblyCopyright("Copyright � Autodesk, Inc 2023")]
```
Comment (reddyashish) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:11]: Don't commit this file

**Comment by QilongTang** [comment]
@reddyashish When are we going to split the follow up improvements into another PR? In general I am scared of huge PRs :)

**Comment by dnenov** [comment]
> 

Yes, it is a big one, isn't it :/ I will check in with @reddyashish to see what's the best way to split it up!

**Comment by QilongTang** [comment]
Thanks @dnenov @reddyashish Maybe we can review this one as it is and leave other improvements on another PR. What do you think?

**Comment by dnenov** [comment]
This can now be closed, all the parts that made up this PR have been successfully merged.

**Pull request closed without merging**